pkg/tuf/client.go

```diff
@@ -540,7 +542,7 @@ func maybeDownloadRemoteTarget(name string, meta data.TargetFileMeta, t *TUF) er
 	// If we already have the target locally, don't bother downloading from remote storage.
 	if cachedTarget, err := t.targets.Get(name); err == nil {
 		// If the target we have stored matches the meta, use that.
-		if isValidTarget(cachedTarget, meta) {
+		if valid, _ := isValidTarget(cachedTarget, meta); valid {
```

Comment:
do we need to check err here and below?

Reply:
I don't think it's needed. This is the `maybeDownloadRemoteTarget`, so it's basically checks if the target exists locally, and that the metadata matches the one from upstream repo. So if the target is not valid, it may be due to a stale cache, so the client will just attempt to download the (updated) target from remote. So I don't think the actual error message is important here, as a stale cache would not necessarily indicate an issue. If there is a "real" issue, such that unsupported hash algorithm or so, the error will bubble up during the next `GetTarget` invocation.